crates/resolver/src/dns_lru.rs

```diff
@@ -34,7 +33,7 @@ use crate::proto::{ProtoError, ProtoErrorKind};
 /// upper bound on received TTLs.
 pub(crate) const MAX_TTL: u32 = 86400_u32;
 
-#[derive(Debug)]
+#[derive(Debug, Clone)]
```

Comment:
Do you understand why this is necessary, conceptually?

Reply:
Moka clones values when returning them, instead of returning a reference, because other threads need to be able to modify the entries after `get()` returns, but before the calling code has dropped the result. I believe it uses crossbeam-epoch internally to manage entry lifetimes, but it closes out its epoch guard after cloning, so that aspect isn't exposed to the user.